macrosAutoCats/src/main/scala/com/softwaremill/macwire/autocats/MacwireAutoCatsMacros.scala

```diff
@@ -144,13 +176,15 @@ object MacwireCatsEffectMacros {
       r
     }
 
-    val generatedInstance = go(targetType.tpe)
+    val resources: Seq[Resource] = values.collect {
+      case r: Resource => r
+      case e: Effect => e.asResource
+    }
 
-    val code =
-      resources.values.foldRight(q"cats.effect.Resource.pure[cats.effect.IO, $targetType]($generatedInstance)") {
-        case (resource, acc) =>
-          q"${resource.value}.flatMap((${resource.ident}: ${resource.tpe}) => $acc)"
-      }
+    val code = resources.foldRight(q"cats.effect.Resource.pure[cats.effect.IO, $targetType](com.softwaremill.macwire.autowire[$targetType](..${values.map(_.ident)}))") {
```

Comment:
hm will this work if the factory method has both a from-resource, and an auto-wire parameter? e.g.

```scala
case class A()
class B
class C
val resourceB: Resource[B] = ...
def makeC(a: A, b: B): Resource[C] = ...

autowire[C](resourceB, makeC _)
```

Reply:
maybe I'm missing some crucial detail of the algorithm, it might be useful to add a comment somewhere with rough steps which would explain what's happening.

Reply:
Yep, good catch, thanks! In that version it was not supported, but the fix was actually pretty straightforward